File:  MVI with Databinding (Java) /app/src/main/java/com/mvvm/mvi/views/LoginBindingData.java
package com.mvvm.mvi.views;

import androidx.databinding.BaseObservable;

import com.mvvm.mvi.BR;

public class LoginBindingData extends BaseObservable {
    public String infoText;
    public boolean isEnableMainFields;
    public boolean isEnableResetButton;
    public int progressBarVisibility;

    public LoginBindingData(String infoText, boolean isEnableMainFields, boolean isEnableResetButton, int progressBarVisibility) {
        this.infoText = infoText;
        this.isEnableMainFields = isEnableMainFields;
        this.isEnableResetButton = isEnableResetButton;
        this.progressBarVisibility = progressBarVisibility;
    }

    public void setData(String infoText, boolean isEnableMainFields, boolean isEnableResetButton, int progressBarVisibility) {
        this.infoText = infoText;
        this.isEnableMainFields = isEnableMainFields;
        this.isEnableResetButton = isEnableResetButton;
        this.progressBarVisibility = progressBarVisibility;
        notifyPropertyChanged(BR._all);
    }
}

File:  MVI with Databinding (Kotlin) /app/src/main/java/com/mvvm/mvi/views/LoginBindingData.kt
package com.mvvm.mvi.views

import androidx.databinding.BaseObservable
import com.mvvm.mvi.BR

class LoginBindingData(var infoText: String, var isEnableMainFields: Boolean,
                       var isEnableResetButton: Boolean, var progressBarVisibility: Int) : BaseObservable() {
    fun setData(infoText: String, isEnableMainFields: Boolean, isEnableResetButton: Boolean, progressBarVisibility: Int) {
        this.infoText = infoText
        this.isEnableMainFields = isEnableMainFields
        this.isEnableResetButton = isEnableResetButton
        this.progressBarVisibility = progressBarVisibility
        notifyPropertyChanged(BR._all)
    }

}
